### PR TITLE
api: replace here-string in `log_diagnose` with process substitution

### PR DESCRIPTION
**Pull request by theofficialgman**
a here-string creates a temporary file on disk which can lead to failure to process all greps in the case of disk space exhaustion. a here-string is also subject to max argument length exhaustion on some systems/kernels leading to failure to process all greps.

process substitution occurs in memory and is not subject to the same limitations.

### CODE REVIEW

**Comment by Botspot** [comment]
> a here-string creates a temporary file on disk

Are you sure?
20241009_21h20m24s_grim

File: api [hunk 1]
```diff
@@ -2858,6 +2858,8 @@ If this error keeps happening, try installing the More RAM app from Pi-Apps. Fin
   #return the information
   local IFS=$'\n'
   echo -e "${error_type}\n${error_caption[*]}"
+
+  rm -f "$tmplogfile"
```
Comment (Botspot) [inline api:2862]: Where is this variable set? I don't see it being assigned anywhere.

Comment (theofficialgman) [reply api:2862]: Sorry. Artifact of a previous refactor attempt.

**Comment by theofficialgman** [comment]
> > a here-string creates a temporary file on disk
> 
> Are you sure?

See top comment https://askubuntu.com/questions/678915/whats-the-difference-between-and-in-bash

**Comment by theofficialgman** [comment]
Also just an FYI regarding ChatGPT and its use for specific technical questions. The use of ChatGPT will not give you consistent good/accurate results. It is a large language model, aka, complicated autocomplete. The question you asked was very leading and didn't provide any background information for it to "attempt" to autocomplete an answer for. It wasn't trained to get the answer correct but instead to answer in a way that sounded "natural". It is unreasonable for you to ask a technical question like that and expect an accurate answer.

Also to provide justification from my statement beyond the askubuntu answer I checked the bash code and quickly found what is responsible for this. The short of it is except in a few specific cases (zero length and short files) a temporary file is made.

https://github.com/bminor/bash/blob/6794b5478f660256a1023712b5fc169196ed0a22/redir.c#L433-L578

**Comment by Botspot** [comment]
Merge at will.

**Pull request merged**